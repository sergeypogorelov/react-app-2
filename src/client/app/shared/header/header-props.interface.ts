export interface HeaderProps {
  search?: string;
}
