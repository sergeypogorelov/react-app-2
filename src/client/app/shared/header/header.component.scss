@import '../../../styles/functions';

$z-index-for-bg: 1;
$z-index-for-content: $z-index-for-bg + 1;

.header {
  &::before {
    content: '';
    display: block;
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: blackColor(0.9);
    z-index: $z-index-for-bg;
  }
  &-logo {
    position: absolute;
    top: 20px;
    left: 60px;
    z-index: $z-index-for-content;
  }
  &-content {
    position: relative;
    z-index: $z-index-for-content;
  }
  padding: 100px 60px;
  position: relative;
  background-image: url('../../../assets/images/cottage.jpg');
}
