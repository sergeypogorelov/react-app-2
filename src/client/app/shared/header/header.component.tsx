import './header.component.scss';

import React, { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';

import { urlFragments } from '../../core/constants/url-fragments';

import { HeaderProps } from './header-props.interface';

import { TitleMain } from '../title-main/title-main.component';

export const Header: FunctionComponent<HeaderProps> = ({
  children,
  search,
  searchShown,
}) => {
  const searchValue = search || '';

  return (
    <div className="header">
      <div className="header-logo">
        <TitleMain />
      </div>
      {searchShown && (
        <Link
          className="header-search"
          to={`/${urlFragments.searchMovie}/${searchValue}`}
        />
      )}
      <div className="header-content">{children}</div>
    </div>
  );
};
