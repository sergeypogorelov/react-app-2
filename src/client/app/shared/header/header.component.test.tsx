import React from 'react';
import renderer from 'react-test-renderer';

import { Header } from './header.component';

test('Header component should render correctly', () => {
  const tree = renderer.create(<Header>test</Header>).toJSON();
  expect(tree).toMatchSnapshot();
});
