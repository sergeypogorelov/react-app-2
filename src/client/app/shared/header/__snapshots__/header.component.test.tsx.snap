// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div
  className="header"
>
  <div
    className="header-logo"
  >
    <p
      className="title-main m-0"
    >
      <span
        className="title-main-prefix"
      >
        netflix
      </span>
      <span>
        roulette
      </span>
    </p>
  </div>
  <div
    className="header-content"
  >
    test
  </div>
</div>
`;
